File: contour_detection/contour_detection.py
import cv2
import numpy as np

cv2.samples.addSamplesDataSearchPath("../test_images")

img_org = cv2.imread(cv2.samples.findFile("pca_test1.jpg"))
img = cv2.blur(img_org, (5,5))
img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
_, img = cv2.threshold(img, 100, 255, cv2.THRESH_BINARY)

contours, heirarchies = cv2.findContours(img, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)

img_out = cv2.drawContours(img_org, contours, -1, (0,0,255), 4)
cv2.imshow("",img_out)
cv2.waitKey()

File: cpp/contour_detection/ContourDetection.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

int main()
{
    cv::samples::addSamplesDataSearchPath("../../test_images");
    cv::Mat img_org = cv::imread(cv::samples::findFile("pca_test1.jpg"));
    cv::Mat img, img_out;
    cv::blur(img_org, img, {5, 5});
    cv::cvtColor(img, img, cv::COLOR_BGR2GRAY);
    cv::threshold(img, img, 100, 255, cv::THRESH_BINARY);
    std::vector<std::vector<cv::Point>> contours;
    std::vector<cv::Vec4i> hierarchy;
    cv::findContours(img, contours, hierarchy, cv::RETR_TREE, cv::CHAIN_APPROX_SIMPLE);
    cv::drawContours(img_org, contours, -1, {0, 0, 255}, 4);
    cv::imshow("Contour Detection", img_org);
    cv::waitKey();
    cv::destroyAllWindows();
}
